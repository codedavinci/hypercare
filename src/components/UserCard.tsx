import { User } from "../types";
import Avatar from "./common/Avatar";
import Button from "./common/Button";
import Ellipsis from "./common/Ellipsis";

type UserCardProps = {
  id: User["id"];
  firstname: User["firstname"];
  lastname: User["lastname"];
  avatar: User["avatar"];
  description: User["description"];
  role: User["role"];
  onViewMore: (userId: string) => void;
};

export default function UserCard({
  id,
  avatar,
  firstname,
  lastname,
  description,
  role,
  onViewMore,
}: UserCardProps) {
  return (
    <div className="card bg-base-100 shadow-xl items-center">
      <div className="pt-6">
        <Avatar
          src={avatar}
          altText={`${firstname} ${lastname}'s profile photo`}
        />
      </div>
      <div className="card-body items-center text-center">
        <h2 className="card-title">
          {firstname} {lastname}
        </h2>
        <div data-theme="light" className="badge badge-primary">
          {role}
        </div>
        <Ellipsis text={description} />
        <div className="card-actions">
          <Button onClick={() => onViewMore(id)} id="view-more">
            View More
          </Button>
        </div>
      </div>
    </div>
  );
}

UserCard.displayName = "UserCard";
