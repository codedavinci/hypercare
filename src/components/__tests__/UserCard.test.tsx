import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import UserCard from "../UserCard";

import RAW_USERS from "../../api/__tests__/usersMock.json";

const SAMPLE_USER = RAW_USERS[0];

describe("UserCard Component", () => {
  it("should render correctly and match snapshot", () => {
    const { asFragment } = render(
      <UserCard
        id={SAMPLE_USER.id}
        firstname={SAMPLE_USER.firstname}
        lastname={SAMPLE_USER.lastname}
        avatar={SAMPLE_USER.avatar}
        description={SAMPLE_USER.description}
        role={SAMPLE_USER.role}
        onViewMore={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onViewMore with correct user ID when "View More" button is clicked', () => {
    const onViewMoreMock = jest.fn();
    const { getByText } = render(
      <UserCard
        id={SAMPLE_USER.id}
        firstname={SAMPLE_USER.firstname}
        lastname={SAMPLE_USER.lastname}
        avatar={SAMPLE_USER.avatar}
        description={SAMPLE_USER.description}
        role={SAMPLE_USER.role}
        onViewMore={onViewMoreMock}
      />
    );
    const viewMoreButton = getByText("View More");
    fireEvent.click(viewMoreButton);
    expect(onViewMoreMock).toHaveBeenCalledWith(SAMPLE_USER.id);
  });
});
