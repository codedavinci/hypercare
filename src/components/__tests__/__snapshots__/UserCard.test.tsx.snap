// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Component should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="card bg-base-100 shadow-xl items-center"
  >
    <div
      class="pt-6"
    >
      <div
        class="avatar"
      >
        <div
          class="w-24 rounded-full ring ring-primary ring-offset-base-100 ring-offset-2"
        >
          <img
            alt="John Doe's profile photo"
            src="avatar1.png"
          />
        </div>
      </div>
    </div>
    <div
      class="card-body items-center text-center"
    >
      <h2
        class="card-title"
      >
        John Doe
      </h2>
      <div
        class="badge badge-primary"
        data-theme="light"
      >
        user
      </div>
      <div
        class="line-clamp-3"
      >
        A sample user
      </div>
      <div
        class="card-actions"
      >
        <button
          class="btn btn-primary"
          id="view-more"
        >
          View More
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
